resource/file/owner/preparer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright © 2016 Asteris, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package owner
+
+import (
+	"strconv"
+
+	"github.com/asteris-llc/converge/load/registry"
+	"github.com/asteris-llc/converge/resource"
+	"golang.org/x/net/context"
+)
+
+// Preparer for Owern
+//
+// Owner sets the file and group ownership of a file or directory.  If
+// `recursive` is set to true and `destination` is a directory, then it will
+// also recursively change ownership of all files and subdirectories.  Symlinks
+// are ignored.
+type Preparer struct {
+	// Destination is the location on disk where the content will be rendered.
+	Destination string `hcl:"destination" required:"true" nonempty:"true"`
+
+	// Recursive indicates whether ownership changes should be applied
+	// recursively.  Symlinks are not followed.
+	Recursive bool `hcl:"recursive" default:"false" export:"recursive"`
+
+	// Username specifies user-owernship by user name
+	Username string `hcl:"user" mutally_exclusive:"uid" export:"user"`
+
+	// UID specifies user-ownership by UID
+	UID *int `hcl:"uid" mutually_exclusive:"user" export:"uid"`
+
+	// Groupname specifies group-ownership by groupname
+	Groupname string `hcl:"group" mutually_exclusive:"gid" export:"gid"`
+
+	// GID specifies group ownership by gid
+	GID *int `hcl:"gid" mutually_exclusive:"group" export:"group"`
+
+	// Verbose specifies that when performing recursive updates, a diff should be
+	// shown for each file to be changed
+	Verbose bool `hcl:"verbose"`
+
+	osProxy OSProxy
+}
+
+// Prepare a new task
+func (p *Preparer) Prepare(ctx context.Context, render resource.Renderer) (resource.Task, error) {
+	var uidStr string
+	var gidStr string
+
+	if p.osProxy == nil {
+		p.osProxy = &OSExecutor{}
+	}
+
+	if nil != p.UID {
+		uidStr = strconv.Itoa(*p.UID)
+	}
+
+	if nil != p.GID {
+		gidStr = strconv.Itoa(*p.GID)
+	}
+
```

Comment:
We shouldn't need to do an explicit check here because we're looking up the username and group name based on the uid/gid and will get an error back if the value is out of range.